MatrixSDK/MXSession.m

```diff
@@ -1718,14 +1720,14 @@ - (void)handleAccountData:(NSDictionary*)accountDataUpdate
                 NSDictionary<NSString*, NSArray<NSString*>*> *directRooms;
                 MXJSONModelSetDictionary(directRooms, event[@"content"]);
 
-                if (directRooms != _directRooms
-                    && ![directRooms isEqualToDictionary:_directRooms])
+                if (directRooms != self.directRooms
```

Comment:
Can we use a temporary variable to host `self.directRooms` to limit atomic synchronisation that can lock the current thread?

Reply:
👍